lib/shopify-cli/ui/strict_spinner.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'cli/ui'
+
+module ShopifyCli
+  module UI
+    module StrictSpinner
+      def self.spin(title, auto_debrief: false)
+        exception = nil
+        CLI::UI::Spinner.spin(title, auto_debrief: auto_debrief) do |*args|
+          begin
+            yield(*args)
+          rescue StandardError => e
+            exception = e
+            CLI::UI::Spinner::TASK_FAILED
+          end
+        end
+        raise exception if exception
```

Comment:
Why did you need to do this? I would prefer that you actually handle these errors in the contextual code, update the spinner with the reason that is failed and abort your command. I know a lot of the command current throw error but stack traces are not a great way to show error states. If you can just abort and display a reason for failure this would make a much better experience

Reply:
Since spinners are concurrent, the CLI::UI opts to capture output and exceptions. This is the only way to access this information from within a spinner right now is through the debrief that outputs stacktrace, stdout, and stderr for the one thread that failed, but that was unacceptable from our UX perspective. We want to provide the user with meaningful errors and only updating the spinner message didn't satisfy that requirement.

The approach used in this class is a common approach in repos that face this issue. Basically this just aims to simulate handling exceptions within a spinner in a single thread. Where we create a StrictSpinner, we can have a rescue block to catch exceptions so we can properly display appropriate error messages - we actually don't show stack traces anywhere in our scripts code right now.

If you have another approach to this problem I'd love to hear it!